elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -288,8 +294,12 @@ public boolean matchesId(String checkId) {
             return Collections.emptySet();
         }
 
+        //filterExpressionCheck
+        FilterExpression permissionFilter = getPermissionFilterExpression(loadClass,
+                ReadPermission.class, requestScope);
+
         Iterable<T> list;
-        FilterScope filterScope = new FilterScope(requestScope, loadClass);
+        FilterScope filterScope = new FilterScope(requestScope, loadClass, permissionFilter);
```

Comment:
I believe this can be removed if we modify the RequestScope.getGlobalFilterExpression function.